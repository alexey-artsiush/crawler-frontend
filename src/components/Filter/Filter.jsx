/* eslint-disable object-curly-newline */
import React, { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import {
  selectFilter,
  setFilterLease,
  setFilterLocation,
  setFilterMaxPrice,
  setFilterMinPrice,
  setFilterRooms,
} from '../../store/filter/filterSlice';
import { Dropdown } from '../Dropdown';
import { Button } from '../Button';
import { getApartment } from '../../store/apartment/apartmentSlice';
import paths from '../../utils/paths';
import './Filter.scss';

export const Filter = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const filter = useSelector(selectFilter);
  const [city, setSelectedLocation] = useState(filter.location);
  const [leavingRoom, setSelectedRooms] = useState(filter.leavingRoom);
  const [rentalPeriod, setSelectedRentalPeriod] = useState(filter.rentalPeriod);
  const [minPrice, setMinPrice] = useState(filter.minPrice);
  const [maxPrice, setMaxPrice] = useState(filter.maxPrice);

  useEffect(() => {
    dispatch(setFilterLocation(city));
    dispatch(setFilterRooms(leavingRoom));
    dispatch(setFilterLease(rentalPeriod));
    dispatch(setFilterMinPrice(minPrice));
    dispatch(setFilterMaxPrice(maxPrice));
  }, []);

  const click = () => {
    dispatch(
      getApartment({
        city,
        leavingRoom,
        rentalPeriod,
        minPrice,
        maxPrice,
      })
    );
    navigate(paths.apartment);
  };

  return (
    <div className="filter">
      <div className="filter-header">
        <div className="filter-banner">
          <h3>Find your apartment here</h3>
          <h4>Apartment rental for short and long term</h4>
        </div>
      </div>
      <div className="filter-panel">
        <div className="filter-point">
          <div className="filter-title">
            <span>Location</span>
          </div>
          <Dropdown
            size="m"
            selected={city}
            setSelected={setSelectedLocation}
            options={[
              'All offers',
              'Minsk',
              'Grodno',
              'Brest',
              'Mogilev',
              'Gomel',
              'Vitebsk',
            ]}
          />
        </div>
        <div className="filter-point">
          <div className="filter-title">
            <span>Rooms</span>
          </div>
          <Dropdown
            size="m"
            selected={leavingRoom}
            setSelected={setSelectedRooms}
            options={['All offers', '1', '2', '3', '4']}
          />
        </div>
        <div className="filter-point">
          <div className="filter-title">
            <span>Lease</span>
          </div>
          <Dropdown
            size="m"
            selected={rentalPeriod}
            setSelected={setSelectedRentalPeriod}
            options={['All offers', 'Short-term', 'Long-term']}
          />
        </div>

        <div className="filter-point">
          <div className="filter-title">
            <span>Price, $</span>
          </div>
          <div className="filter-point-wrapper">
            <div className="filter-point">
              <div className="filter-point-price dropdown dropdown--medium">
                <input
                  type="number"
                  onChange={(e) => setMinPrice(e.target.value)}
                  placeholder="from"
                />
                <input
                  type="number"
                  onChange={(e) => setMaxPrice(e.target.value)}
                  placeholder="to"
                />
              </div>
            </div>
          </div>
        </div>

        <div className="filter-dropdown-wrapper">
          <div className="filter-wrapper">
            <Button
              size="l"
              type="orange"
              onClick={() => {
                click();
              }}
            >
              Search
            </Button>
          </div>
        </div>
      </div>
    </div>
  );
};
