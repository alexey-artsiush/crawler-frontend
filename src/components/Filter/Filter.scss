@import '../../utils/colors.scss';

.filter {
  background-color: $c-blue;
  color: $c-white;
  display: flex;
  flex-wrap: wrap;
  margin: -3px 0 140px;
}

.filter-dropdown-wrapper {
  display: flex;
  justify-content: space-around;
  width: 100%;
}

.filter-panel {
  background-color: $c-white;
  border-radius: 8px;
  box-shadow: 1px 8px 30px 0 $c-grey-light;
  color: $c-blue-dark;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  margin: -69px auto;
  padding: 18px 22px;
  width: 50%;
}

.filter-point {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  width: 21%;
  min-width: 155px;
}

.filter-point-price {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  width: 60%;

  input {
    background-color: #fff;
    border: none;
    margin: -10px 0 5px;
    text-align: center;
  }
}

.filter-point-wrapper {
  display: flex;
  justify-content: space-around;
  align-items: center;
  width: 100%;
}

.filter-point-wrapper-item {
  min-width: 45%;
}

.filter-header {
  padding-bottom: 7vw;
  width: 100%;
}

.filter-banner {
  margin-left: 20vh;
}

.filter-title {
  display: flex;
  font-size: 18px;
  font-weight: 600;
  justify-content: center;
  margin-bottom: 5px;
  width: 100%;
}

.filter-wrapper {
  margin-left: auto;
  margin-right: 0;
}

@media (max-width: 1478px) {
  .filter {
    margin-bottom: 100px;
  }

  .filter-point,
  .filter-title {
    display: flex;
    justify-content: center;
  }

  .filter-header {
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin-bottom: 60px;
  }

  .filter-panel {
    flex-wrap: wrap;
  }
}

@media (max-width: 1050px) {
  .filter-panel {
    flex-wrap: wrap;
    width: 80%;
  }

  .filter-header {
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin-bottom: 60px;
  }

  .filter-banner {
    margin-left: 2vh;
  }
}

@media (max-width: 790px) {
  .filter-point {
    width: 35%;
  }
}

@media (max-width: 690px) {
  .filter-dropdown-wrapper {
    flex-direction: column;
  }

  .filter-point-wrapper {
    flex-direction: column;
  }
}

@media (max-width: 500px) {
  .filter-dropdown-wrapper {
    align-items: center;
    max-width: 90%;
  }

  .filter-wrapper {
    margin: 0;
  }
}
