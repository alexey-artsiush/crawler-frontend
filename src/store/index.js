/* eslint-disable implicit-arrow-linebreak */
import { configureStore } from '@reduxjs/toolkit';
import apartmentSlice from './apartment/apartmentSlice';
import premiumApartmentSlice from './premiumApartment/premiumApartmentSlice';
import userApartmentSlice from './userApartment/userApartmentSlice';
import filterSlice from './filter/filterSlice';
import userSlice from './user/userSlice';
import newsSlice from './news/newsSlice';
import citySlice from './city/citySlice';
import chatArticleSlice from './chat/chatArticleSlice';

const store = configureStore({
  reducer: {
    chat: chatArticleSlice,
    user: userSlice,
    filter: filterSlice,
    apartment: apartmentSlice,
    premiumApartment: premiumApartmentSlice,
    userApartment: userApartmentSlice,
    news: newsSlice,
    city: citySlice,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
