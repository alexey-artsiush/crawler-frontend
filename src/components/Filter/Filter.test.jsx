/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { Filter } from './Filter';
import store from '../../store/index';

describe('Filter component tests', () => {
  it('Render brand phrase', () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Filter />
        </Provider>
      </BrowserRouter>
    );
    const helloElement = screen.getByText(/Find you apartment here/i);
    expect(helloElement).toBeInTheDocument();
  });
  it('Filter component snapshot', () => {
    const filter = render(
      <BrowserRouter>
        <Provider store={store}>
          <Filter />
        </Provider>
      </BrowserRouter>
    );
    expect(filter).toMatchSnapshot();
  });
});
