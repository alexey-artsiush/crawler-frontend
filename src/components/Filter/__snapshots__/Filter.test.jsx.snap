// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component tests Filter component snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filter"
      >
        <div
          class="filter-header"
        >
          <div
            class="filter-banner"
          >
            <h3>
              Find you apartment here
            </h3>
            <h4>
              Apartment rental for short and long term
            </h4>
          </div>
        </div>
        <div
          class="filter-panel"
        >
          <div
            class="filter-point"
          >
            <div
              class="filter-title"
            >
              <span>
                Location
              </span>
            </div>
            <div
              class="dropdown dropdown--medium"
            >
              <div
                aria-hidden="true"
                class="dropdown-btn"
              />
            </div>
          </div>
          <div
            class="filter-point"
          >
            <div
              class="filter-title"
            >
              <span>
                Rooms
              </span>
            </div>
            <div
              class="dropdown dropdown--medium"
            >
              <div
                aria-hidden="true"
                class="dropdown-btn"
              />
            </div>
          </div>
          <div
            class="filter-point"
          >
            <div
              class="filter-title"
            >
              <span>
                Lease
              </span>
            </div>
            <div
              class="dropdown dropdown--medium"
            >
              <div
                aria-hidden="true"
                class="dropdown-btn"
              />
            </div>
          </div>
          <div
            class="filter-wrapper"
          >
            <button
              class="btn btn--orange btn--medium undefined"
              type="button"
            >
              Search
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filter"
    >
      <div
        class="filter-header"
      >
        <div
          class="filter-banner"
        >
          <h3>
            Find you apartment here
          </h3>
          <h4>
            Apartment rental for short and long term
          </h4>
        </div>
      </div>
      <div
        class="filter-panel"
      >
        <div
          class="filter-point"
        >
          <div
            class="filter-title"
          >
            <span>
              Location
            </span>
          </div>
          <div
            class="dropdown dropdown--medium"
          >
            <div
              aria-hidden="true"
              class="dropdown-btn"
            />
          </div>
        </div>
        <div
          class="filter-point"
        >
          <div
            class="filter-title"
          >
            <span>
              Rooms
            </span>
          </div>
          <div
            class="dropdown dropdown--medium"
          >
            <div
              aria-hidden="true"
              class="dropdown-btn"
            />
          </div>
        </div>
        <div
          class="filter-point"
        >
          <div
            class="filter-title"
          >
            <span>
              Lease
            </span>
          </div>
          <div
            class="dropdown dropdown--medium"
          >
            <div
              aria-hidden="true"
              class="dropdown-btn"
            />
          </div>
        </div>
        <div
          class="filter-wrapper"
        >
          <button
            class="btn btn--orange btn--medium undefined"
            type="button"
          >
            Search
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
